### PR TITLE
Upgrade old Python version in packaging pipeline

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

- Upgrade from Python 3.6 to 3.8 in packaging pipeline.
- Raise build.py minimum required Python version.

### Motivation and Context

Stop using unsupported Python version.

### CODE REVIEW

File: build.amd64.1411.bat
Comment (skottmckay) [inline build.amd64.1411.bat:1]: We could probably delete this file - it is for an ancient VS toolset

File: tools/ci_build/github/linux/docker/inference/aarch64/default/cpu/scripts/install_deps.sh [hunk 1]
```diff
@@ -62,5 +63,15 @@ fi
 GetFile https://nodejs.org/dist/v16.14.2/node-v16.14.2-linux-${NODEJS_ARCH}.tar.gz /tmp/src/node-v16.14.2-linux-${NODEJS_ARCH}.tar.gz
 tar --strip 1 -xf /tmp/src/node-v16.14.2-linux-${NODEJS_ARCH}.tar.gz -C /usr
 
+echo "Installing Python"
```
Comment (skottmckay) [inline tools/ci_build/github/linux/docker/inference/aarch64/default/cpu/scripts/install_deps.sh:66]: nit: saying why we're doing it manually in the comment would be good

File: build.amd64.1411.bat
Comment (edgchen1) [reply build.amd64.1411.bat:1]: cool, will do

**Review by thiagocrepaldi — COMMENTED** [review]
Just as an extra data point, but the new ONNX exporter based on PyTorch Dynamo has a better model coverage with Python 3.11

It would be nice to start testing this environment in our pipelines for the near future exporter

**Comment by edgchen1** [comment]
> Just as an extra data point, but the new ONNX exporter based on PyTorch Dynamo has a better model coverage with Python 3.11
> 
> It would be nice to start testing this environment in our pipelines for the near future exporter

Good to know. FYI @microsoft/onnxruntime-es

**Review by skottmckay — APPROVED** [review]

**Comment by snnn** [comment]
@thiagocrepaldi , the pipelines he is modifying is not a python pipeline. It is for generating nuget packages. Our python packaging pipeline supports python 3.11.

**Pull request merged**